src/java/org/apache/cassandra/batchlog/BatchlogManager.java

```diff
@@ -459,36 +458,31 @@ private void writeHintsForUndeliveredEndpoints(int startFrom, Set<InetAddressAnd
             Keyspace keyspace = Keyspace.open(ks);
             Token tk = mutation.key().getToken();
 
+
             ReplicaLayout.ForTokenWrite liveAndDown = ReplicaLayout.forTokenWriteLiveAndDown(keyspace, tk);
             Replicas.temporaryAssertFull(liveAndDown.all()); // TODO in CASSANDRA-14549
 
-            EndpointsForToken.Builder liveReplicasBuilder = EndpointsForToken.builder(tk);
+            Replica selfReplica = liveAndDown.all().selfIfPresent();
+            if (selfReplica != null)
```

Comment:
What exactly is this corner case where we have a hint written to ourselves? I thought we don't hint ourselves.